hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketAcceptorThread.java

```diff
@@ -77,46 +92,93 @@ public void run() {
             logger.finest("Starting SocketAcceptor on " + serverSocketChannel);
         }
 
-        Selector selector = null;
+        if (selectorWorkaround) {
```

Comment:
I would do this coordinated at a single place. Otherwise you will get 3+3+1 of these log entries.